src/App.tsx

```diff
@@ -73,45 +87,85 @@ class Board extends React.Component<BoardProps, BoardState> {
         });
     }
 
+    public cornerMarkSelectedSquares(i: SquareValue) {
+        const newMarkingMap = new Map(this.state.markingMap);
+        for (const address of this.state.highlights) {
+            if (!this.isPermanent(address) && this.state.values[address] == null) {
```

Comment:
Invert this if condition and have the body of the if have `continue`. This will reduce the level of indentation. In general you should always try to return/continue early in the function/loop and have the body be at the lowest indentation level possible.